astropy/cosmology/parameter/_core.py

```diff
@@ -70,6 +70,9 @@ class Parameter:
 
     Parameters
     ----------
+    default : Any or `NotImplemented` (optional, keyword-only)
```

Comment:
I find this rather strange because `Any` already includes `NotImplemented`. 
```suggestion
    default : Any (optional, keyword-only)
```

Reply:
Sure. It was to distinguish valid values (All) from the invalid one (NotImplemented), but yes the former is technically a superset of the latter.

Reply:
See #15400